src/TestResult.dfy

```diff
@@ -49,4 +51,93 @@ module TestResult {
     var resourceCount :- Externs.ParseNat(resourceCountStr);
     Success(TestResult(displayName, outcome, durationTicks, resourceCount))
   }
-}
\ No newline at end of file
+
+  method GroupTestResults(results: seq<TestResult>) returns (groupedResults: map<string, seq<TestResult>>) {
+    groupedResults := map[];
+    for resIndex := 0 to |results| {
+      var res := results[resIndex];
+      var name := res.displayName;
+      if name in groupedResults {
+        groupedResults := groupedResults[name := groupedResults[name] + [res]];
+      } else {
+        groupedResults := groupedResults[name := [res]];
+      }
+    }
+  }
+
+  function method TestResultStatistics(results: seq<TestResult>, f: TestResult -> int): Statistics.Statistics
```

Comment:
It probably makes more sense to have `f` be of type `TestResult -> real`, and convert from ticks to seconds _before_ calculating statistics for duration measurements. But `Min`, `Max`, and `Map` are specific to `seq<int>`, so we'd have to introduce custom versions of them.

Reply:
Why not change Min, Max, and Map to work with reals and do everything in terms for reals?

Reply:
Or are they library functions?  In that case we need MaxF that takes a comparison function